src/app.types.tsx

```diff
@@ -48,10 +51,37 @@ export interface RecordMetadata {
   activeExperiment?: string;
 }
 
-export interface ChannelMetadata {
-  dataType: Scalar | Image | Waveform;
+export type DataType = 'scalar' | 'image' | 'waveform';
+
+export interface FullCommonChannelMetadata {
+  systemName: string;
+  dataType: DataType;
+  userFriendlyName?: string;
+  description?: string;
+  units?: string;
+}
+
+export interface FullScalarChannelMetadata extends FullCommonChannelMetadata {
+  dataType: 'scalar';
+  sf?: number;
```

Comment:
Honestly this took me a minute to get but now I realise it denotes significant figures. Could we call it `significantFigures` for better clarity?

Reply:
ah yeah - I didn't think about that - I was too involved in rounding algorithms that `sf` made total sense to me - `significantFigures` is much better. `sigFigs` is another shorthand used in the requirement but I think it's not too long that spelling it out is horrendous